tests/callbacks_test.go

```diff
@@ -601,3 +601,174 @@ func (s *CallbacksSuite) TestNexusResetWorkflowWithCallback() {
+	 * 2. Start WF second time w/ a callback
+	 * 3. Reset WF back to the first run
+	 * 4. Verify callback is called
+	 */
+
+	tv := testvars.New(s.T())
+	ctx := testcore.NewContext()
+	sdkClient, err := client.Dial(client.Options{
+		HostPort:  s.FrontendGRPCAddress(),
+		Namespace: s.Namespace().String(),
+	})
+	s.NoError(err)
+
+	taskQueue := tv.TaskQueue()
+	workflowID := tv.WorkflowID()
+
+	ch := &completionHandler{
+		requestCh:         make(chan *nexus.CompletionRequest, 2),
+		requestCompleteCh: make(chan error, 2),
+	}
+	callbackAddress := fmt.Sprintf("localhost:%d", freeport.MustGetFreePort())
+	s.runNexusCompletionHTTPServer(s.T(), ch, callbackAddress)
+
+	w := worker.New(sdkClient, taskQueue.GetName(), worker.Options{})
+
+	w.RegisterWorkflow(blockingWorkflow)
+	s.NoError(w.Start())
+	defer w.Stop()
+
+	// 1. Start WF w/ no callbacks and immediately terminate
+	request1 := &workflowservice.StartWorkflowExecutionRequest{
+		RequestId:          uuid.NewString(),
+		Namespace:          s.Namespace().String(),
+		WorkflowId:         workflowID,
+		WorkflowType:       &commonpb.WorkflowType{Name: "blockingWorkflow"},
+		TaskQueue:          taskQueue,
+		Input:              nil,
+		WorkflowRunTimeout: durationpb.New(20 * time.Second),
+		Identity:           s.T().Name(),
+	}
+
+	startResponse1, err := s.FrontendClient().StartWorkflowExecution(ctx, request1)
+	s.NoError(err)
+
+	// Validate the workflow started, then terminate it
+	workflowExecution := &commonpb.WorkflowExecution{
+		WorkflowId: workflowID,
+		RunId:      startResponse1.RunId,
+	}
+	s.WaitForHistoryEvents(`
+			1 WorkflowExecutionStarted
+			2 WorkflowTaskScheduled
+			3 WorkflowTaskStarted
+			4 WorkflowTaskCompleted`,
+		s.GetHistoryFunc(s.Namespace().String(), workflowExecution),
+		5*time.Second,
+		10*time.Millisecond)
+
+	_, err = s.FrontendClient().TerminateWorkflowExecution(ctx, &workflowservice.TerminateWorkflowExecutionRequest{
+		Namespace:         s.Namespace().String(),
+		WorkflowExecution: workflowExecution,
+		Reason:            s.T().Name(),
+		Identity:          tv.WorkerIdentity(),
+	})
+	s.NoError(err)
+
+	s.EqualHistoryEvents(`
+			1 WorkflowExecutionStarted
+			2 WorkflowTaskScheduled
+			3 WorkflowTaskStarted
+			4 WorkflowTaskCompleted
+			5 WorkflowExecutionTerminated`,
+		s.GetHistoryFunc(s.Namespace().String(), workflowExecution)())
+
+	// 2. Start WF second time w/ callbacks (new run)
+	cbs := []*commonpb.Callback{
+		{Variant: &commonpb.Callback_Nexus_{Nexus: &commonpb.Callback_Nexus{Url: "http://" + callbackAddress + "/cb1"}}},
+	}
+
+	request2 := proto.Clone(request1).(*workflowservice.StartWorkflowExecutionRequest)
+	request2.RequestId = uuid.NewString()
+	request2.CompletionCallbacks = cbs
+
+	startResponse2, err := s.FrontendClient().StartWorkflowExecution(ctx, request2)
+	s.NoError(err)
+	s.True(startResponse2.Started)
+	s.NotEqual(workflowExecution.RunId, startResponse2.RunId)
+
+	s.WaitForHistoryEvents(`
```

Comment:
Not required, redundant, you don't care about this workflow making progress at all. Just reset it wherever it is.